diff --git a/core/definitions.mk b/core/definitions.mk
index 11cea3f..c1854e3 100644
--- a/core/definitions.mk
+++ b/core/definitions.mk
@@ -1298,7 +1298,7 @@ $(hide) if [ -d "$(PRIVATE_SOURCE_INTERMEDIATES_DIR)" ]; then \
 fi
 $(hide) tr ' ' '\n' < $(dir $(PRIVATE_CLASS_INTERMEDIATES_DIR))/java-source-list \
     | sort -u > $(dir $(PRIVATE_CLASS_INTERMEDIATES_DIR))/java-source-list-uniq
-$(hide) $(TARGET_JAVAC) -encoding ascii $(PRIVATE_BOOTCLASSPATH) \
+$(hide) $(TARGET_JAVAC)  $(PRIVATE_BOOTCLASSPATH) \
     $(addprefix -classpath ,$(strip \
         $(call normalize-path-list,$(PRIVATE_ALL_JAVA_LIBRARIES)))) \
     $(PRIVATE_JAVACFLAGS) $(strip $(PRIVATE_JAVAC_DEBUG_FLAGS)) $(xlint_unchecked) \
diff --git a/core/product_config.mk b/core/product_config.mk
index 70ac894..78422c5 100644
--- a/core/product_config.mk
+++ b/core/product_config.mk
@@ -231,6 +231,9 @@ PRODUCT_LOCALES := $(strip \

 # Everyone gets nodpi assets which are density-independent.
 PRODUCT_LOCALES += nodpi
+PRODUCT_LOCALES += hdpi
+PRODUCT_LOCALES += xhdpi
+

 # Assemble the list of options.
 PRODUCT_AAPT_CONFIG := $(PRODUCT_LOCALES)

